### ISSUE TITLE
AMICAL with VISIR data

### ISSUE THREAD

**Issue by stellaCL** [SydneyAstrophotonicInstrumentationLab/AMICAL#130]
Hello AMICAL developers!

As I noted on ESO's VISIR website, this code is the proposed tool for SAM data reduction. But is there a manual for it? I saw there is an example tutorial just for SPHERE.

Apologies in advance if this question should not be on Issues.

Thank you for your time !

**Comment by DrSoulain** [comment]
Yes, VISIR is compatible with AMICAL but we did not test yet this feature. It's supposed similar to what is done for SPHERE actually, so feel free to try it on your own dataset.

**Comment by DrSoulain** [comment]
Hi @stellaCL, did you manage to solve your problem? This issue can be closed for now?

**Comment by bentitan** [comment]
Hello everyone,

I have some visir SAM data to reduce and I want to test the amical code, Which are the input data requeriments? 

First, I compared with JWST data and those FITS files are in data cubes, but my raw observations are distributed on diferents frames. Do I need to create my own cube to make the reduction? Which is the structure of the input data?

My second question is about the header. Do amical use information from the header? Which Keywords? But more important, do amical recognize VISIR header?

**Comment by DrSoulain** [comment]
Hi @bentitan, usually, the appropriate data format is datacube in fits files. One file is typically a bunch of frames used to compute the statistics and build the covariance matrices (uncertainties). You can check, for instance, the example files used for SPHERE or NIRISS. AMICAL does use the header keyword that should be standard between ESO instruments (at least). You can follow the tutorial used for SPHERE dataset, I should be very similar to the VISIR one.

**Comment by bentitan** [comment]
Hello @DrSoulain 

Thanks for your answers. After all I clean all my observations manually to recover the raw interferograms using sigma clipping. Then I follow the same CHOP/NOD steps of calibration:

Achop - Bchop    -    (Achop -Bchop)
       NOD1                        NOD2

I cropped the images, turn the negatives into positives multiplying by -1. Finally I put all my data into a cube of dimensions of x/y/N, where x and y are the number of pixels in x and y directions and N is the amount of interferograms that I obtained and I modified the header a little.

After that I tried example_SPHERE.py changing the respective input parameters for VISIR and the code partially works, but I am not sure if the output data is completely good because the visibility is not well behaved.

Since I do not understand very good the inner reduction of amical, I have two possible ideas why I have this problem.

1. VISIR/SAM mode does not have an exposure calulcator to obtain a good SNR, therefore the integration time of our observations is not enough to calculate the observables. Actually I cannot run amical with allowing _clip=True_

2. On one of my diagnostic plots the splodge positions looks very shifted respect to the power spectrum and it cannot do the fit well
![Figure_10](https://github.com/SAIL-Labs/AMICAL/assets/132116301/8942d4fa-a9f4-4de8-b7d9-648306b889d4)

 I should modify my input parameters but I do not kwow which one. 

Do you have any idea how to fix it? (As a comment, I use exactly the same input file _example_SPHERE.py_, but with my data)
